quoridor-logic-server/src/main/java/com/rl/mpquoridor/models/Board.java

```diff
@@ -0,0 +1,187 @@
+            System.out.println("");
+            for (int width = 0; width < this.board[0].length; width++) {
+                System.out.print(this.board[height][width] + " , ");
+            }
+        }
+
+        System.out.println("");
+    }
+
+    public boolean movePawn(Position fromPosition, Position toPosition) {
+        //TODO: check if this is a valid new position
+        this.board[fromPosition.getY() * 2][fromPosition.getX() * 2] = Cell.EMPTY;
+        this.board[toPosition.getY() * 2][toPosition.getX() * 2] = Cell.PAWN;
+        return true;
+    }
+
+    public boolean addWall(Wall wall, Position[] positions, int[] heightsWin) {
+        if (this.checkAddingWall(wall, positions, heightsWin)) {
+            this.board = this.addWallToBoard(wall, this.board);
+            return true;
+        }
+
+        return false;
+    }
+
+    public boolean checkAddingWall(Wall wall, Position[] positions, int[] heightsWin) {
+        Direction direction = wall.getDirection();
+        Position position = wall.getPosition();
+        int newX = position.getX() * 2 + 1;
+        int newY = position.getY() * 2 + 1;
+
+        if (direction == Direction.Right) {
+            if (newX + 2 > this.board[0].length) {
+                return false;
+            } else {
+                if (this.board[newY][newX] == Cell.WALL ||
+                        this.board[newY][newX + 1] == Cell.WALL ||
+                        this.board[newY][newX + 2] == Cell.WALL) {
+                    return false;
+                }
+            }
+        } else {
+            if (newY + 2 > this.board.length) {
+                return false;
+            } else {
+                if (this.board[newY][newX] == Cell.WALL ||
+                        this.board[newY + 1][newX] == Cell.WALL ||
+                        this.board[newY + 2][newX] == Cell.WALL) {
+                    return false;
+                }
+            }
+        }
+
+        return this.checkPaths(wall, positions, heightsWin);
+    }
+
+    private void initBoard() {
+        for (int height = 0; height < this.board.length; height++) {
+            for (int width = 0; width < this.board[0].length; width++) {
+                this.board[height][width] = Cell.EMPTY;
+            }
+        }
+    }
+
+    private boolean checkPaths(Wall wall, Position[] positions, int[] heightsWin) {
+        Cell[][] newBoard = new Cell[this.board.length][this.board[0].length];
+        for (int height = 0; height < this.board.length; height++) {
+            System.arraycopy(this.board[height], 0, newBoard[height], 0, this.board[0].length);
+        }
+
+        newBoard = this.addWallToBoard(wall, newBoard);
+
+        for (int i = 0; i < positions.length; i++) {
+            if (this.checkingExistingPath(newBoard, positions[i], heightsWin[i])) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    //BFS
```

Comment:
I'm sure that there is existing library which implements BFS
mayby there is something easy to use